### DISCUSSION TITLE
Problems with getStaticProps() and next/link.

### DISCUSSION THREAD

**Discussion by frames75** [vercel/next.js#24213] [Help]
I've been practicing with `getStaticProps()` and `getServerSideProps()` for a while.

Using `getStaticProps()`, *revalidate* prop and 'next/link' to create client routing, I've found that once built the app. and open any of these pages, these are never updated in client-side using that Links, no matter the API data changes and the page is updated in server-side.

Links seem to always open the cached version of the page, and you must press F5 to load the new version updated in server-side.

Using \<a\> tag instead of Link, `getStaticProps()` and *revalidate* prop works like a charm, but you lose the maravellous UX that Link provides.

Am I missing anything? It would be great to be able to use `getStaticProps()`, *revalidate* prop and Links together.

----

[Next.js Docs](https://nextjs.org/docs/basic-features/data-fetching#statically-generates-both-html-and-json) say:

> **Statically Generates both HTML and JSON**
<br>When a page with getStaticProps is pre-rendered at build time, in addition to the page HTML file, Next.js generates a JSON file holding the result of running getStaticProps.
<br>This JSON file will be used in client-side routing through 'next/link' or 'next/router'. When you navigate to a page that’s pre-rendered using getStaticProps, Next.js fetches this JSON file (pre-computed at build time) and uses it as the props for the page component. This means that client-side page transitions will not call getStaticProps as only the exported JSON is used.
<br>When using Incremental Static Generation getStaticProps will be executed out of band to generate the JSON needed for client-side navigation. You may see this in the form of multiple requests being made for the same page, however, this is intended and has no impact on end-user performance

In my experience, the last paragraph is wrong and page is never updated in client-side (using 'next/link', using \<a\> tag works).

**Comment by Emiliano-Bucci** [comment]
I'm facing the same issue; any ideas? :/
